.github/workflows/build-and-integration-tests.yml

```diff
@@ -122,30 +115,112 @@ jobs:
         ./${{ matrix.output }} binstall --manifest-path . --no-confirm cargo-binstall
         cargo binstall --help
 
-    - name: Create archive (tgz, linux)
-      if: ${{ matrix.os != 'macos-latest' && matrix.os != 'windows-latest' }}
-      run: tar -czvf cargo-binstall-${{ matrix.target }}.tgz ${{ matrix.output }}
-
-    - name: Create archive (zip, windows)
-      if: ${{ matrix.os == 'windows-latest' }}
-      run: tar.exe -a -c -f cargo-binstall-${{ matrix.target }}.zip ${{ matrix.output }}
-
-    - name: Create archive (zip, macos)
-      if: ${{ matrix.os == 'macos-latest' }}
-      run: zip cargo-binstall-${{ matrix.target }}.zip ${{ matrix.output }}
+    - name: Upload output
+      uses: actions/upload-artifact@v3
+      with:
+        retention-days: 1
+        name: "${{ matrix.target }}.${{ matrix.output }}"
+        path: "${{ matrix.output }}"
+
+  release-test:
+    name: Package and release (testing)
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Get outputs
+      uses: actions/download-artifact@v3
+      with:
+        path: outputs/
 
-    - name: Upload artifacts
-      uses: actions/upload-artifact@v1
+    - name: Create archives
```

Comment:
I think we can extract this to a script under `ci-script`?
I've seen the same code twice.

Reply:
Oh, my bad, the duplicated section shouldn't have been left here